content/platforms/kubernetes/db-controller.md

```diff
@@ -0,0 +1,271 @@
+        name: rec
+    ```
+
+1. Apply the file to your namespace containing your cluster and the operator:
+
+    ```sh
+    kubectl apply -f db.yaml
+    ```
+
+1. Check for the status of your database:
+
+    ```sh
+    kubectl get redb/mydb -o yaml
+    ```
+
+    Once the status is `active`, the database is ready for use.
+
+{{% note %}}
+The `jq` tool is a useful tool for checking the status via the json output:
+
+```sh
+kubectl get redb/mydb -o json | jq .status.status
+```
+
+which will output the status value from all the JSON returned.
+
+{{% /note %}}
+
+
+### Modifying databases
+
+A database is modified by changing the custom resource for the database. You
+can simply edit your original specification and apply the change or use
+kubectl edit:
+
+1. Edit the definition:
+
+    ```sh
+    kubectl edit redb/mydb
+    ```
+
+2. Modify the specification (only properties in `spec` section) and save the changes.
+
+3. Monitor the status to see when the changes become effective:
+
+    ```sh
+    kubectl get redb/mydb -o yaml
+    ```
+
+    Once the status is `active`, the database is again ready for use.
+
+### Deleting databases
+
+A database is deleted by deleting the custom resource for the database. The
+database controller will remove the database from the cluster and the various
+services will also be removed:
+
+```sh
+kubectl delete redb/mydb
+```
+
+
+## Connecting to database
+
+Once a database is created via the database controller, the service are
+created in the same namespace that can be used to access the database.
+Connection information for the database is stored in a secret. The name of
+that secret is stored specification or was specified as part of the
+database custom resource.
+
+This secret contains:
+
+ * the database port (port)
+ * the database service name (service_name)
+ * the database password for authenticating (password)
+
+The database controller maintains these connection values in the secret. An
+application can use this secret in a variety of ways. A simple way is to map
+them to environment variables in a deployment pod.
+
+For example, we can setup a test pod to access a database `redb/mydb`:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: database-client
+spec:
+  containers:
+    - name: test-container
+      image: redis
+      command: [ "sleep" ]
+      args: [ "300" ]
+      env:
+      - name: REDIS_PORT
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: port
+      - name: REDIS_HOST
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: service_name
+      - name: REDIS_AUTH
+        valueFrom:
+          secretKeyRef:
+            name: redb-mydb
+            key: password
+
+```
+
+Then we can connect to our database via exec:
+
+```sh
+kubectl exec -it database-client -- /bin/bash -c "redis-cli -h \$REDIS_HOST -p \$REDIS_PORT -a \$REDIS_AUTH"
+```
+
+
+## Options for databases
```

Comment:
We have API documentation in the docs repo (generated from the internal repo).
It's fine to have this here as well, but we should have a mechanism, manual or automatic, to keep these in sync.
https://github.com/RedisLabs/redis-enterprise-operator/blob/master/deploy/redis_enterprise_database_api.md